danger_code_ws/2.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <cstdlib>
+
+// Function to dynamically allocate a matrix of size rows x cols
+double** allocateMatrix(int rows, int cols) {
+    double** matrix = new double*[rows];
+    for (int i = 0; i < rows; i++) {
+        matrix[i] = new double[cols];
+    }
+    return matrix;
+}
+
+// Function to initialize a matrix with random values
+void initializeMatrix(double** matrix, int rows, int cols) {
+    for (int i = 0; i < rows; i++) {
+        for (int j = 0; j < cols; j++) {
+            matrix[i][j] = (std::rand() % 100) / 10.0;  // Random values between 0 and 10
```

Comment:
While using `std::rand()` for testing purposes is acceptable, it is not suitable for production-level matrix operations. For better performance and reproducibility, use a higher-quality random number generator like those found in `<random>`.  Consider seeding the generator for reproducibility.